src/v/utils/extended_rwlock.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "seastarx.h"
+#include "ssx/semaphore.h"
+#include <seastar/core/rwlock.hh>
+
+class extended_rwlock;
+
+class rwlock_unit {
```

Comment:
if you plan on keeping this version (instead of patching seastar), can you please add some unit tests for this utility?